velox/vector/BaseVector.h

```diff
@@ -177,6 +177,14 @@ class BaseVector {
     return type_;
   }
 
+  void setType(TypePtr type) {
+    VELOX_CHECK_NOT_NULL(type);
+    VELOX_CHECK(
+        type_->kindEquals(type),
+        "The set type should be kindEquals() with type_ except names");
```

Comment:
Let's make sure to include current and new type in the error message to help debugging and re-write the error message for readability.

"Cannot change vector type from {} to {}. The old and new types can be different logical types, but the underlying physical types must match."

Reply:
@mbasmanova Updated.